virttest/qemu_vm.py

```diff
@@ -3561,34 +3561,38 @@ def activate_nic(self, nic_index_or_name):
         if nic['nic_model'] == 'virtio-net-pci':
             if int(nic['queues']) > 1:
                 device_add_cmd += ",mq=on"
-            if 'vectors' in nic:
-                device_add_cmd += ",vectors=%s" % nic.vectors
+                if 'vectors' in nic:
+                    device_add_cmd += ",vectors=%s" % nic.vectors
         device_add_cmd += nic.get('nic_extra_params', '')
         if 'romfile' in nic:
             device_add_cmd += ",romfile=%s" % nic.romfile
         bus = self.devices.get_buses({'aobject': 'pci.0'})[0]
-        if isinstance(bus, qdevices.QPCIEBus):
-            root_port_id = bus.get_free_root_port()
-            if root_port_id:
-                device_add_cmd += ",bus=%s" % root_port_id
-                root_port = self.devices.get_buses({'aobject': root_port_id})[0]
-                root_port.insert(qdevices.QBaseDevice(nic.nic_model,
-                                                      aobject=nic.nic_name))
-            else:
-                raise virt_vm.VMAddNicError("No free root port for device %s"
-                                            " to plug." % nic.nic_name)
-        error_context.context("Activating nic on VM %s with monitor command %s" % (
-            self.name, device_add_cmd))
+        nic_dev = qdevices.QDevice(params={'id': nic.nic_name,
```

Comment:
Please double check this update not impact ppc platform.

Reply:
@ngu-niny  can you help to review this patch?

Reply:
I focus on ppc, this update is suitable ‘virtio-net’ for ppc and x86, 'spapr-vlan' does not support hotplug, so it does not use this.

Reply:
thanks to known.